modules/c++/math.linear/include/math/linear/Line2D.h

```diff
@@ -0,0 +1,250 @@
+/* =========================================================================
+ * This file is part of math.linear-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * math.linear-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <types/RowCol.h>
+#include <cmath>
+#include <except/Exception.h>
+#include <iostream>
+
+#ifndef __MATH_LINEAR_LINE_2D_H__
+#define __MATH_LINEAR_LINE_2D_H__
+
+namespace math
+{
+namespace linear
+{
+typedef types::RowCol<double> Point2D;
+
+class Line2D
```

Comment:
I think all the methods should move to the cpp as they're unlikely to be inlined and make the API harder to read